File: machine/cpp/render/gx/view_snapshot.cpp
#include "render/gx/view_snapshot.h"

#include "machine/devices/gx/device_output.h"
#include "render/gameview.h"

namespace bmsx {

void commitGxGpuViewSnapshot(GameView& view, const GxGpuDeviceOutput& output) {
	view.gxGpuCommandBuffer = &output.commandBuffer;
	view.gxGpuReadbackPort = &output.readbackPort;
	view.gxGpuStatusWord = output.statusWord;
	view.gxGpuDisplayModeWord = output.displayModeWord;
	view.gxGpuDisplayStartWord = output.displayStartWord;
	view.gxGpuHorizontalDisplayRangeWord = output.horizontalDisplayRangeWord;
	view.gxGpuVerticalDisplayRangeWord = output.verticalDisplayRangeWord;
	view.gxGpuVramSnapshotBytes = &output.vramSnapshotBytes;
	view.gxGpuVramSnapshotSerial = output.vramSnapshotSerial;
	view.gxCharacterPlaneOutput = &output.characterPlane;
}

} // namespace bmsx

File: machine/ts/render/gx/view_snapshot.ts
import type { GxGpuDeviceOutput } from '../../machine/devices/gx/device_output';
import type { GameView } from '../gameview';

export function commitGxGpuViewSnapshot(view: GameView, output: GxGpuDeviceOutput): void {
	view.gxGpuCommandBuffer = output.commandBuffer;
	view.gxGpuReadbackPort = output.readbackPort;
	view.gxGpuStatusWord = output.statusWord;
	view.gxGpuDisplayModeWord = output.displayModeWord;
	view.gxGpuDisplayStartWord = output.displayStartWord;
	view.gxGpuVramYAddressExtensionWord = output.vramYAddressExtensionWord;
	view.gxGpuHorizontalDisplayRangeWord = output.horizontalDisplayRangeWord;
	view.gxGpuVerticalDisplayRangeWord = output.verticalDisplayRangeWord;
	view.gxGpuPcrtcScanout = output.pcrtcScanout;
	view.gxGpuPcrtcScanoutRevision = output.pcrtcScanout.revision;
	view.gxGpuVramSnapshotBytes = output.vramSnapshotBytes;
	view.gxGpuVramSnapshotSerial = output.vramSnapshotSerial;
	view.gxGpuVramReplacementSerial = output.vramReplacementSerial;
}
